gdal/frmts/vrt/vrtmultidim.cpp

```diff
@@ -1328,7 +1339,7 @@ bool VRTMDArraySourceInlinedValues::Read(const GUInt64* arrayStartIdx,
         {
             // For negative step request, temporarily simulate a positive step
             // and fix up the start at the end of the loop.
-            start_i = start_i + (count[i]-1) * step_i;
+            start_i = start_i - (count[i]-1) * (-step_i);
```

Comment:
Think comments are good here because if I saw this I might want to simplify by reworking the algebra.

Reply:
Same as an above occurence. Comment added